src/assets/scripts/client/airport/MapCollection.js

```diff
@@ -0,0 +1,140 @@
+import _forEach from 'lodash/forEach';
+import _has from 'lodash/has';
+import _isArray from 'lodash/isArray';
+import _isObject from 'lodash/isObject';
+import BaseCollection from '../base/BaseCollection';
+import MapModel from './MapModel';
+
+/**
+ * Collection of `MapModel` objects
+ *
+ * Provides methods to create `MapModel` objects, used by `AirportModel`
+ * and to extract video map line data to be rendered by the
+ * `CanvasController`
+ *
+ * @class MapCollection
+ * @extends BaseCollection
+ */
+export default class MapCollection extends BaseCollection {
+    /**
+     * @constructor
+     * @param mapJson {object}
+     * @param airportPositionModel {StaticPositionModel}
+     * @param magneticNorth {number}
+     */
+    constructor(mapJson, airportPositionModel, magneticNorth) {
+        super();
+
+        if (!_isArray(mapJson) && !_isObject(mapJson)) {
+            throw new TypeError(
+                `Invalid parameter passed to MapCollection. Expected an array or object but found ${typeof mapJson}`
+            );
+        }
+
+        /**
+         * @inherited
+         * @memberof BaseCollection
+         * @property _items
+         * @type {array<MapModel>}
+         * @default []
+         */
+
+        /**
+         * @inherited
+         * @memberof BaseCollection
+         * @property length
+         * @type {number}
+         * @default #_items.length
+         */
+
+        /**
+         * @property _airportPositionModel
+         * @type {StaticPositionModel}
+         * @default null
+         * @private
+         */
+        this._airportPositionModel = null;
+
+        /**
+         * @property _magneticNorth
+         * @type {number}
+         * @default null
+         * @private
+         */
+        this._magneticNorth = null;
+
+        this._init(mapJson, airportPositionModel, magneticNorth);
+    }
+
+    /**
+     * Public fascade for `#_items`
+     *
+     * @property maps
+     * @return {array<MapModel>}
+     */
+    get maps() {
+        return this._items;
+    }
+
+    /**
+     * @for MapCollection
+     * @method getVisibleMapLines
+     * @return {array<array<object>>}
```

Comment:
does this actually return a 2d array?  or is it actually `array<MapModel>`?

Reply:
As is, it does return a 2d array . The rationale was that it returns all the lines that are to be displayed, but I wanted to avoid the overhead of merging the array every time the canvas is redrawn. I've only just realised that the videomap isn't collected every time an aircraft position changes.

The other reason I don't actually use it was because if the UI is ever updated to allow the user to select which maps are rendered, I wasn't sure if this would be done by filtering the MapModels in the MapCollection, or in the CanvasController.

Using this method, and doing the reduction in the MapsCollection would certainly make the CanvasController cleaner

Reply:
Cool.

We've tried to encapsulate as much logic and _"work"_ as possible within these models and collections as possible.  That way the controllers can (in theory) be lighter.  They can concentrate on operating on the data, and not have to worry about assembling any of it.

In some places we are better about this than in others, but thats the general intent: MVC.